src/storage/storage.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * User: llukac<lukas@lightstreams.io>
+ * Date: 11/12/19 10:10
+ * Copyright 2019 (c) Lightstreams, Granada
+ */
+
+const Web3Wrapper = require('../web3');
+
+const aclSc = require('../../build/contracts/ACL.json');
+
+/**
+ * Uploads a new file into Smart Vault.
+ *
+ * @param from {string} Address paying ACL deployment fee
+ * @param owner {string} Address of the owner of the file in SC
+ * @param file {ReadableStream|File} File to add
+ * @param isPublic {boolean} Whenever everyone can read the file content
+ *
+ * @returns Object{acl: "0xF1E3dC4E704D18783fF58b1f1217f2d82f0b9544", meta: "QmVd6sTUugEUugimMNWLcczhgN3zxunRQmriYQzzi9bRzV"}
+ */
+module.exports.add = async (web3, gatewayStorage, { from, owner, file, isPublic = false }) => {
```

Comment:
IMO passing the `gatewayStorage` instead of the gateway object itself feels like a halfway between specialitation, what means passing directly the `addWithAcl` method call, and the generalization, what should be to pass the `gateway` object.

Reply:
As we discussed, passing the whole Gateway object doesn't make sense because it has many more modules which are not being used. Passing just specifically gateway object API is more efficient and also easier to test/mock generally work with.